Upstream: yes, master branch

From 3350f5e2bab646e0c3e8cf221a3b668a688eb3a0 Mon Sep 17 00:00:00 2001
From: Michael Jansen <kde@michael-jansen.biz>
Date: Wed, 16 Oct 2013 23:26:11 +0200
Subject: [PATCH] FFMPEG Renamed enum CodecID to AVCodedID

And all values got an AV_ prefix. I found the exact version number with
google. So not sure its correct.
---
 CMakeLists.txt                              |  9 ++++++++-
 plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp | 14 ++++++++++----
 plugins/decoder/ffmpeg/k3bffmpegwrapper.h   |  2 --
 3 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca1531d..a3e047e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,7 +147,14 @@ if(K3B_BUILD_FFMPEG_DECODER_PLUGIN)
       #else
       #include <ffmpeg/avcodec.h>
       #endif
-      int main() { enum CodecID t = CODEC_ID_MP3; return 0; }
+      int main() {
+      #if LIBAVCODEC_BUILD >= AV_VERSION_INT(54,25,0)
+      # Not needed here. AV_CODEC_ID_MP3 exists.
+      #else
+      enum CodecID t = CODEC_ID_MP3;
+      #endif
+      return 0;
+      }
       " HAVE_FFMPEG_CODEC_MP3)
 
     cmake_pop_check_state()
diff --git a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
index a5ef087..3bd2d54 100644
--- a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
+++ b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -208,14 +208,20 @@ int K3bFFMpegFile::type() const
 
 QString K3bFFMpegFile::typeComment() const
 {
+#if LIBAVCODEC_BUILD < AV_VERSION_INT(54,25,0)
+    #define AV_CODEC_ID_WMAV1  CODEC_ID_WMAV1
+    #define AV_CODEC_ID_WMAV2  CODEC_ID_WMAV2
+    #define AV_CODEC_ID_MP3    CODEC_ID_MP3
+    #define AV_CODEC_ID_AAC    CODEC_ID_AAC
+#endif
     switch( type() ) {
-    case CODEC_ID_WMAV1:
+    case AV_CODEC_ID_WMAV1:
         return i18n("Windows Media v1");
-    case CODEC_ID_WMAV2:
+    case AV_CODEC_ID_WMAV2:
         return i18n("Windows Media v2");
-    case CODEC_ID_MP3:
+    case AV_CODEC_ID_MP3:
         return i18n("MPEG 1 Layer III");
-    case CODEC_ID_AAC:
+    case AV_CODEC_ID_AAC:
         return i18n("Advanced Audio Coding (AAC)");
     default:
         return QString::fromLocal8Bit( d->codec->name );
diff --git a/plugins/decoder/ffmpeg/k3bffmpegwrapper.h b/plugins/decoder/ffmpeg/k3bffmpegwrapper.h
index 1ec36b6..d1671c8 100644
--- a/plugins/decoder/ffmpeg/k3bffmpegwrapper.h
+++ b/plugins/decoder/ffmpeg/k3bffmpegwrapper.h
@@ -18,8 +18,6 @@
 
 #include "k3bmsf.h"
 
-
-
 /**
  * Create with K3bFFMpegWrapper::open
  */
-- 
1.8.4.3

